mbuild/recipes/polymer.py

```diff
@@ -68,6 +73,16 @@ def __init__(self, monomers, n, sequence='A', port_labels=('up', 'down')):
         # Hoist the first part's bottom port to be the bottom port of the polymer.
         self.add(first_part.labels[port_labels[1]], port_labels[1], containment=False)
 
+        # Add a cap to left/right or both ends of the pattern i.e. C->ABABAB
+        for cap, c_port, end, e_port in zip(caps,cap_ports,
+        [first_part,last_part],[1,0]):
+            if cap != None:
+                assert_port_exists(c_port, cap)
```

Comment:
We should have an assertion earlier in the method that checks that `len(caps) == len(cap_ports)`